hack/determine_install_upgrade_version_test.py

```diff
@@ -0,0 +1,95 @@
+		self.assertEqual(get_matching_versions('origin', self.test_pkgs_objs, '1.2'), ['1.2.0-0.el7.x86_64', '1.2.5-0.el7.x86_64'])
+
+	def test_get_matching_versions_only_release(self):
+		""" when both release and pre-release versions exist, only release versions are returned """
+		self.assertEqual(get_matching_versions('origin', self.test_pkgs_objs, '1.4'), ['1.4.1-1.el7.x86_64'])
+		self.assertEqual(get_matching_versions('origin', self.test_pkgs_objs, '1.3'), ["1.3.1-1.el7.x86_64", "1.3.2-3.el7.x86_64"])
+
+class DetermineSearchVersionTestCase(unittest.TestCase):
+	"Test for `determine_install_upgrade_version.py`"
+
+	def test_determine_search_for_origin_standard_versioning_version(self):
+		""" when the origin version is higher then the first version of the new origin versioning schema - origin-3.6 """
+		self.assertEqual(determine_search_version("origin", "3.7.0"), "3.6")
+
+	def test_determine_search_for_origin_standard_to_legacy_versioning_version(self):
+		""" when the origin version is the first from the new origin versioning schema - origin-3.6 """
+		self.assertEqual(determine_search_version("origin", "3.6.0"), "1.5")
+
+	def test_determine_search_for_origin_legacy_version(self):
+		""" when the origin version is in the old versioning schema """
+		self.assertEqual(determine_search_version("origin", "1.5.0"), "1.4")
+
+	def test_determine_search_for_openshift_ansible_standard_versioning_version(self):
+		""" when openshift-ansible, which doesnt have different versioning schema, is in 3.7 version  """
+		self.assertEqual(determine_search_version("openshift-ansible", "3.7.0"), "3.6")
+
+	def test_determine_search_for_openshift_ansible_standard_to_legacy_versioning_version(self):
+		""" when openshift-ansible, which doesnt have different versioning schema is in 3.6 version """
+		self.assertEqual(determine_search_version("openshift-ansible", "3.6.0"), "3.5")
+
+	def test_determine_search_for_openshift_ansible_legacy_version(self):
+		""" when openshift-ansible, which doesnt have different versioning schema is in 3.4 version """
+		self.assertEqual(determine_search_version("openshift-ansible", "3.5.0"), "3.4")
+
+class GetInstallUpgradeVersionTestCase(unittest.TestCase):
+	"Test for `determine_install_upgrade_version.py`"
+
+	def test_get_install_upgrade_with_multiple_matching_release_versions(self):
+		""" when install and upgrade version are picked from multiple matching versions of released versions are """
+		matching_versions = ["1.2.0-1.el7.x86_64", "1.2.2-1.el7.x86_64", "1.2.5-1.el7.x86_64"]
+		install_upgrade_versions = ["1.2.0-1.el7.x86_64", "1.2.5-1.el7.x86_64"]
+		self.assertEqual(get_install_upgrade_version(matching_versions), install_upgrade_versions)
+
+	def test_get_install_upgrade_with_single_matching_release_version(self):
+		""" when install and upgrade version are picked from single matching versions of released versions are """
```

Comment:
hm -- very hard to see the difference. I'd suggest one of two things:

 - your method names / test names are verbose and expressive. if you think you captured the entirety of the test reason in the name, get rid of the comment
 - emphasize the differences... so this might read "when only a single matching version is present..."